src/omnicore/tx.cpp

```diff
@@ -411,30 +622,84 @@ bool CMPTransaction::interpret_MetaDExCancelEcosystem()
 /** Tx 50 */
 bool CMPTransaction::interpret_CreatePropertyFixed()
 {
-    if (pkt_size < 25) {
-        return false;
+    if (encodingClass == OMNI_CLASS_D) {
+        int i = 0;
+
+        std::vector<uint8_t> vecVersionBytes = GetNextVarIntBytes(i);
+        std::vector<uint8_t> vecTypeBytes = GetNextVarIntBytes(i);
+
+        memcpy(&ecosystem, &pkt[i], 1);
+        i++;
+
+        std::vector<uint8_t> vecPropTypeBytes = GetNextVarIntBytes(i);
+        std::vector<uint8_t> vecPrevPropIdBytes = GetNextVarIntBytes(i);
+
+        const char* p = i + (char*) &pkt;
+        std::vector<std::string> spstr;
+        for (int j = 0; j < 5; j++) {
+            spstr.push_back(std::string(p));
+            p += spstr.back().size() + 1;
+        }
+
+        if (isOverrun(p)) {
+            PrintToLog("%s(): rejected: malformed string value(s)\n", __func__);
+            return false;
+        }
+
+        int j = 0;
+        memcpy(category, spstr[j].c_str(), std::min(spstr[j].length(), sizeof(category)-1)); j++;
```

Comment:
I don't think this would offer much near term (as you couldn't fit >255 chars of metadata in a Class D anyway without support from a miner) but the current default nulldata limit may not always be the same so I do see why it might be worth doing...

In that case do you think we'd look at it as a malformed transaction and drop it completely (rather than parsing it and just marking it invalid)?

P.S. personally I'd like to abandon category and subcategory completely (I did in the Litecoin port hehe) and restrict the number of characters for the remaining metadata but that's a change that probably belongs in a different PR.